Fix multi-project builds

This happens in some multi-project builds:

```
Failed to execute goal com.lightbend.rp:reactive-app-maven-plugin:0.3.0:build (build-docker) on project hello-impl: Execution build-docker of goal com.lightbend.rp:reactive-app-maven-plugin:0.3.0:build failed: java.lang.ClassNotFoundException: com.lightbend.lagom.internal.api.tools.ServiceDetector$
```

Seems to be related to lagom endpoint discovery magic.  

This error should be occuring on the [maven chirper](https://github.com/lagom/lagom-java-maven-chirper-example/) project

Thanks @mitkus for the comment and running me through this on Hangout.

Here are the action items to complete this:

- [ ] Add api-tools dependency checking logic, don't throw `java.lang.ClassNotFoundException: com.lightbend.lagom.internal.api.tools.ServiceDetector$`
- [ ] Mention the api-tools dependency in the README, where the other dependencies are mentioned
- [ ] Exclude `lagom-javadsl-client` in https://github.com/lightbend/reactive-app-maven-plugin/blob/d6ce0f3969632019dcda9321fab75d19d4e32aa8/src/main/java/com/lightbend/rp/AppTypeDetector.java#L18 to avoid:

```
    Caused by: java.lang.ClassCastException: class play.api.inject.guice.GuiceApplicationLoader
        at java.lang.Class.asSubclass (Class.java:3404)
        at com.lightbend.lagom.internal.api.tools.ServiceDetector$.services (ServiceDetector.scala:58)
        at com.lightbend.lagom.internal.api.tools.ServiceDetector$.services (ServiceDetector.scala:51)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at com.lightbend.rp.LagomApp.apply (LagomApp.java:89)
```

--- a/README.md
+++ b/README.md
@@ -82,7 +82,8 @@ important steps in starting up your application successfully. You can add these 
 ```
 
 Plugin will do its best to complain if you should depend on something and forgot to add it. Rule of thumb is: if you use Akka Cluster features, you
-need `reactive-lib-akka-cluster-bootstrap`; if you use Lagom, you also need `reactive-lib-service-discovery`.
+need `reactive-lib-akka-cluster-bootstrap`; if you use Lagom, you also need `reactive-lib-service-discovery` and
+`api-tools`.
 
 ## Building a docker image
 

--- a/src/main/java/com/lightbend/rp/BuildMojo.java
+++ b/src/main/java/com/lightbend/rp/BuildMojo.java
@@ -76,6 +76,9 @@ public class BuildMojo extends AbstractMojo {
             throw new MojoExecutionException("Unknown app type");
 
         settings.read(pluginConf);
+
+        checker.preApplyCheck();
+
         analyser.apply(mavenProject);
 
         // Add default application if analysers didn't create any
@@ -90,7 +93,6 @@ public class BuildMojo extends AbstractMojo {
                     "do that by adding <mainClass>com.app.Main</mainClass> to plugin configuration");
         }
 
-        // TODO: check reactive-lib deps here
         checker.check();
 
         Xpp3Dom conf = configuration(

--- a/src/main/java/com/lightbend/rp/ReactiveLibChecker.java
+++ b/src/main/java/com/lightbend/rp/ReactiveLibChecker.java
@@ -60,6 +60,26 @@ public class ReactiveLibChecker {
         return found;
     }
 
+    public void preApplyCheck() throws MojoExecutionException {
+        boolean success = true;
+
+        switch(settings.appType) {
+            case Basic:
+                break;
+            case Akka:
+                break;
+            case Play:
+                break;
+            case Lagom:
+                success = success && require("api-tools", reactiveLibVersion);
+                break;
+        }
+
+        if (!success) {
+            throw new MojoExecutionException("Library dependency requirements not met.");
+        }
+    }
+
     public void check() throws MojoExecutionException {
         boolean success = true;
         if (settings.enableCommon) {

